tests/test_data.py

```diff
@@ -44,6 +44,14 @@ def test_titanic_class(self, mock_get, mock_select):
             )
         )
 
+        titanic.solve_feature_selection(k=3, alpha=0.5, solver="cqm")
+        mock_select.assert_called_with(num_features=3, alpha=0.5, solver="cqm")
+        mock_get.assert_called_with(
+            mock_select(num_features=3, alpha=0.5).fit_transform(
+                titanic.X.values, titanic.y
+            )
+        )
```

Comment:
You should add each separate test within a subtest to test them separately.
```suggestion
        with self.subTest("nl"):
            titanic.solve_feature_selection(k=3, alpha=0.5, solver="nl")
            mock_select.assert_called_with(num_features=3, alpha=0.5, solver="nl")
            mock_get.assert_called_with(
                mock_select(num_features=3, alpha=0.5).fit_transform(
                    titanic.X.values, titanic.y
                )
            )
```
Similarly for "cqm" below.

Reply:
Also, the below asserts would only test for CQM here, no (on lines 58-59 and 62-63) since that's executed latest, no? If so they should probably also be placed within the subtests above.

Reply:
Yes, even better. :)